clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -11004,8 +11004,10 @@ def err_sycl_restrict : Error<
   "nor constant-initialized"
   "}0">;
 def warn_sycl_kernel_too_big_args : Warning<
-  "size of kernel arguments (%0 bytes) exceeds supported maximum of %1 bytes "
-  "on GPU">, InGroup<SyclStrict>;
+  "size of kernel arguments (%0 bytes) may exceed supported maximum "
+  "on some devices">, InGroup<SyclStrict>;
+def note_sycl_kernel_too_big_args : Note<"supported maximum on some devices "
+  "is %0 bytes">;
```

Comment:
What was the reason to split the diagnostic?  Suggesting:
"size of kernel arguments (%0 bytes) may exceed the supported maximum of %1 bytes on some devices"

Reply:
I wasn't sure if that sounded like 2048 is the supported maximum for SYCL. We need the message to be clear that this argument size may not work but it also could if you aren't using a device with this limitation. I was confused by English language :) I can change it to suggested diagnostic if you prefer that.